sktime/transformers/series_as_features/rocket.py

```diff
@@ -37,7 +37,8 @@ class Rocket(BaseSeriesAsFeaturesTransformer):
     def __init__(self, num_kernels=10_000, normalise=True, random_state=None):
         self.num_kernels = num_kernels
         self.normalise = normalise
-        self.seed = random_state if isinstance(random_state, int) else None
+        self.random_state = \
```

Comment:
Could we simply assign the random state here, whatever it is, and then you [`check_random_state`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html) from scikit-learn in the `_generate_kernels_function`?

Reply:
Everything else looks good! 🙂

Reply:
Please open a separate PR for the random state stuff. 

The `check_random_state` returns a random state object, so you can use that directly to use Numpy's random functionality, e.g.

rng = check_random_state(self.random_state)
subsample = rng.randint(self.n_instances, size=subsample_size)

Reply:
So the reason I'm not doing this at the moment is that you can't pass a numpy random state object into numba compiled code.  I don't think there's a way around it at the moment.  It's possible to pass in an integer to seed numba's own random state (what I'm doing here), but that's it.

Reply:
Okay, I remember talking about this before, let's leave it as it is :)